src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,423 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import ring_theory.algebraic
+import topology.algebra.polynomial
+import analysis.calculus.mean_value
+import data.real.irrational
+import data.set.intervals.infinite
+import algebra.big_operators
+
+open_locale big_operators
+open polynomial real set finset
+
+/-!
+# Liouville's theorem
+
+This file contains the proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+-/
+
+/--
+- a number is transcendental ↔ a number is not algebraic
+- a Liouville's number `x` is a number such that
+  for every natural number `n`, there is a `a/b ∈ ℚ - ℤ` such that `0 < |x - a/b| < 1/bⁿ`
+-/
+
+def liouville_number (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
+  1 < b ∧ 0 < abs(x - a / b) ∧ abs(x - a / b) < 1/b^n
+
+namespace liouville
+
+lemma not_liouville_zero : ¬ liouville_number 0 :=
+begin
+  classical,
+  intro rid,
+  rcases rid 1 with ⟨a, b, b_gt_1, abs_sub_pos, abs_sub_small⟩,
+  simp only [one_div, zero_sub, abs_neg, pow_one] at abs_sub_pos abs_sub_small,
+  set a' := abs a,
+  have a'_ineq : 0 ≤ a' := abs_nonneg _,
+  have : abs (a' / b : ℝ) = abs (a / b), simp only [int.cast_abs, abs_div, abs_abs],
+  rw ←this at abs_sub_pos abs_sub_small,
+  rw [abs_of_nonneg, inv_eq_one_div, div_lt_div_iff, one_mul, mul_lt_iff_lt_one_left] at abs_sub_small,
+  norm_cast at abs_sub_small,
+  have : a' = 0, linarith,
+  simp only [this, int.cast_zero, zero_div, abs_zero] at abs_sub_pos, linarith,
+  repeat { norm_cast, linarith },
+  { apply div_nonneg, norm_cast, assumption, norm_cast, linarith }
+end
+
+lemma abs_eval_rat_ge_one_div_denom_pow_n (f : polynomial ℤ) (a b : ℤ)
+  (b_pos : 0 < b) (not_root : (f.map (algebra_map ℤ ℝ)).eval (a/b) ≠ 0) :
+  1/(b : ℝ)^f.nat_degree ≤ abs ((f.map (algebra_map ℤ ℝ)).eval (a/b)) :=
+have b_nonzero : b ≠ 0, from ne_of_gt b_pos,
+have pos : 0 < 1 / (b : ℝ) ^ f.nat_degree, from
+begin
+  refine div_pos (show 0 < (1 : ℝ), from by linarith) (pow_pos _ _),
+  norm_cast, exact b_pos,
+end,
+have eq₁ : (f.map (algebra_map ℤ ℝ)).eval (a/b) =
+  1 / b ^ f.nat_degree * ∑ (i : ℕ) in f.support, f.coeff i * a ^ i * b ^ (f.nat_degree - i), from
+calc ((f.map (algebra_map ℤ ℝ)).eval (a/b))
```

Comment:
1) These proofs need to be reformatted according to the style guide.
2) I hope they can be split into smaller reusable components (that should probably move to other places in mathlib).

Reply:
Please refer commit afff4c3, to see whether any improvement has been made regarding 1 if not 2.